src/plugins/intel_npu/src/backend/src/zero_init.cpp

```diff
@@ -201,8 +200,44 @@ ZeroInitStructsHolder::ZeroInitStructsHolder() : log("NPUZeroInitStructsHolder",
     // Get our target device
     zeroUtils::throwOnFail("zeDeviceGet", zeDeviceGet(driver_handle, &device_count, &device_handle));
 
+    // Create context from backend to be shared with compiler
     ze_context_desc_t context_desc = {ZE_STRUCTURE_TYPE_CONTEXT_DESC, 0, 0};
     zeroUtils::throwOnFail("zeContextCreate", zeContextCreate(driver_handle, &context_desc, &context));
+
+    // Allocate memory for the char array
+    graphExtName = new char[graph_ext_name.size() + 1];
```

Comment:
why do we need this allocation here?

Reply:
`graphExtName` is non-const char* class member, where `graph_ext_name.size()` will return a const char *.
i have try `graphExtName = const_cast<char*>(graph_ext_name.c_str());` but it lead to different result,
so i use this approach allocation and `strcpy` over the content, and free up `std::string graph_ext_name` instead

Reply:
Can we change that const then? use only char there?

Reply:
More than that we need this only for a log in the compiler side. Can we remove it at all?

Reply:
i checked, i think this allocation might not require, i will remove it in next commit